scripts/core/selfssl

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+##--------------------------------------------------------------------
+## Copyright (c) 2017 OSIsoft, LLC
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+##--------------------------------------------------------------------
+
+
+# @date: 2018-01-02
+#
+# Bash Script to create a new self-signed SSL Certificate
+#
+# USAGE: this command will ask for the certificate name and number in days it will expire
+# $ chmod +x selfssl
+# $ ./selfssl
+#
+# OPTIONAL: run the command
+# $ ./selfssl foglamp 365
+
+# Create a certificate key.
+# Create the certificate signing request (CSR) which contains details such as the domain name and address details.
+# Sign the certificate
+# Install the certificate and key in the application.
+
+
+
+# Default directory to place the Certificate
+
+# SSL_DIR="${FOGLAMP_ROOT}/etc/certs"
+
+SSL_NAME=$1
+SSL_EXPIRATION_DAYS=$2
+
+# Change to your company details
+country=US
+state=California
+# locality=
+organization=OSIsoft
+# organizational_unit=
+common_name=FogLAMP
+email=info@dianomic.com
+
+
+# add fail_if_error and log
+
+if [ -z $1 ]; then
+  printf "Enter the SSL Certificate Name:"
+  read SSL_NAME
```

Comment:
I don't think we want to have interactive use like this - FogLAMP is meant to be standalone/unattended. This script will be used during installation and during an Admin API request, but not be a user sitting at a terminal.